databases/sqlite/plugins.nix

```diff
@@ -0,0 +1,132 @@
+{ lib
+, callPackage
+, fetchFromGitHub
+, fetchzip
+}:
+
+let
+
+  mkSqliteExt =
+    { name
+    , sourcesFn
+    , outFn
+    }:
+
+    { lib
+    , sqlite
+    , stdenv
+    }:
+    let
+      outFile = outFn (if stdenv.isDarwin then "dylib" else "so");
+
+      EXT_DIR = "lib/sqlite/ext/";
+
+      sqliteSrc =
+        let
+          version = "3.39.3";
+          src = fetchzip {
+            name = "sqlite-${version}-source";
+            url = "https://www.sqlite.org/src/tarball/sqlite.tar.gz?r=release";
```

Comment:
This needs a version

Reply:
Also explain why I need to download the source, separately (and can't just use `sqlite.src`.